lib/lumberjack/server.rb

```diff
@@ -214,13 +226,14 @@ def initialize(fd)
       super()
       @parser = Parser.new
       @fd = fd
+      @closed = Concurrent::AtomicBoolean.new
```

Comment:
Oh I see what you mean here, I was doing it in the lumberjack input itself, see 
https://github.com/logstash-plugins/logstash-input-lumberjack/pull/53/files#diff-1b9590423b15f04f215635164e7376ecR98

I will change that, I will make the `@closed` from the server a shared states among the connections.
`Connection#close` will only close the current connection and will not change the value of this variable.

The read loop will check against that value, if it should be closed I will call `#close` on the connection socket.

I am pretty sure that if you close the `tcp_server` socket in the acceptor thread it will make the connection thread raise an IOError.